gemstone/src/swapper/models.rs

```diff
@@ -63,12 +66,22 @@ pub struct GemSwapQuote {
     pub from_amount: String,
     pub to_amount: String,
     pub provider: GemProviderData,
-    pub approval: Option<GemApprovalData>,
+    pub approval: GemApprovalType,
+    pub request: GemSwapRequest,
+}
+
+#[derive(Debug, Clone, uniffi::Enum)]
+pub enum GemApprovalType {
+    Approve(GemApproveData),
+    Permit2(GemApproveData),
+    None,
 }
 
 #[derive(Debug, Clone, uniffi::Record)]
-pub struct GemApprovalData {
+pub struct GemApproveData {
+    pub token: String,
     pub spender: String,
+    pub amount: String,
```

Comment:
```suggestion
    pub value: String,
```

Reply:
This is ERC20 approve naming, let's keep it?